src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexNode.cs

```diff
@@ -220,6 +224,11 @@ bool WithCache(uint context)
                         is_nullable = _alts.IsNullableFor(context);
                         break;
 
+                    case SymbolicRegexKind.OrderedOr:
```

Comment:
Another thing that is for sure something that should be done outside of this PR, but we should probably should unify with the none-symbolic engines and use `RegexNodeKind` enum for these instead.

Reply:
Sounds like a good idea. The `RegexNodeKind` has some entries that aren't valid for `SymbolicRegexNode`, so some extra checks on the symbolic side might be in order, and some new node kinds in `RegexNodeKind` would be required.

Reply:
There is some overlap but also some differences between `RegexNodeKind`  and `SymbolicRegexKind`, 
(e.g. `Not`, `And`, if they are going to remain, but also other cases such as `EndAnchorZRev` that is `EndAnchorZ` when reversed, `WatchDog`, and some cases in `RegexNodeKind` that are not distinguished in `SymbolicRegexKind` concerning for example different kinds of loops and their loop bodies)